pkg/storage/cacher.go

```diff
@@ -359,20 +424,40 @@ func (c *Cacher) Codec() runtime.Codec {
 }
 
 func (c *Cacher) processEvent(event watchCacheEvent) {
+	indexValue, supported := "", false
+	if c.indexFunc.IndexName != "" {
+		var err error
+		indexValue, err = c.indexFunc.IndexFunc(event.Object)
+		if err == nil {
+			supported = true
+		}
```

Comment:
There is one issue with this code now - in case of updates, we need to evaluate index for both old a new value and send this event for watchers for both values. This isn't difficult to add, but it's missing from here.